examples/website/scenegraph-layer/app.js

```diff
@@ -45,12 +49,32 @@ export class App extends Component {
     this._loadData();
   }
 
+  componentWillUnmount() {
+    if (this.state.nextTimeoutId) {
+      clearTimeout(this.state.nextTimeoutId);
+    }
+  }
+
+  _sort(data, oldData) {
+    // In order to keep the animation smooth we need to always return the same
+    // objects in the exact same order. This function will discard new objects
+    // and only update existing ones.
+    if (!oldData) {
+      return data;
+    }
+
+    const dataAsObj = {};
+    data.forEach(entry => (dataAsObj[entry[DATA_INDEX.UNIQUE_ID]] = entry));
+    return oldData.map(entry => dataAsObj[entry[DATA_INDEX.UNIQUE_ID]] || entry);
+  }
+
   _loadData() {
     fetch(DATA_URL)
       .then(resp => resp.json())
       .then(resp => {
         if (resp && resp.states) {
-          this.setState({data: resp.states});
+          const nextTimeoutId = setTimeout(() => this._loadData(), REFRESH_TIME);
+          this.setState({data: this._sort(resp.states, this.state.data), nextTimeoutId});
```

Comment:
This will fail if the app has already been unmounted.

Reply:
But I cancel the timeout in that case.

Reply:
If your promise is resolved after the app is unmounted,

- The timeout is not cancelled because it's not set yet
- Calling `this.setState` will throw

Reply:
How do I prevent that?

Reply:
Set a flag when the app unmounts.

Reply:
ok 🎏

Reply:
Done